### PR TITLE
Add custom webmock test to verify post request

### PR DESCRIPTION
**Pull request by bootstraponline**
This style of testing is interesting because all the internals can change without breaking the behavior (`POST` request).

Also it caught a bug that regular unit tests wouldn't. The symbol for dSYM was wrong. As a result, only the file name was posted instead of the actual contents of the dSYM file.

### CODE REVIEW

File: lib/hockey_gerrit.rb [hunk 1]
```diff
@@ -1,103 +1,104 @@
...
+      options[:dsym_filename] = dsym
+    else
+      is_android = File.extname(ipa) == '.apk'
+      puts 'dSYM not found! Unable to symbolicate crashes' unless is_android
     end
+    options
+  end
+
+  def hockey_url(response)
+    config_url = response.body ? response.body['config_url'] : ''
```
Comment (rolyatwilson) [inline lib/hockey_gerrit.rb]: If response.body is falsey and config_url is set to an empty string, won't that just cause other things to break down the line?

Comment (bootstraponline) [inline lib/hockey_gerrit.rb]: Good catch, fixed.

File: spec/helper/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  def self.disable_net
+    File.exist?(upload_request_path)
+  end
+
+  def post_url # used by webmock
+    'https://upload.hockeyapp.net/api/2/apps/upload'
+  end
+
+  def stub_valid_obj
+    config_url = '{ "config_url": "https://upload.hockeyapp.net/manage/apps/123456/app_versions/9"}'
```
Comment (rolyatwilson) [inline spec/helper/spec_helper.rb]: Tests depend on this url (with upload changed to rink). Would it be better to declare 'upload_url' and 'rink_url' variables so we are not using strings in the tests?

File: spec/unit_test_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require_relative 'helper/spec_helper'
+
+describe HockeyGerrit do
+  let(:hockey) { HockeyGerrit.new }
+
+  it '#git_log' do
+    log = hockey.git_log
+    raise 'git_log fail' if log.empty?
```
Comment (rolyatwilson) [inline spec/unit_test_spec.rb]: why not "expect { ... }.to_not ?

File: spec/unit_test_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  it '#git_commit_sha' do
+    log = hockey.git_commit_sha
+    raise 'git_commit_sha fail' if log.empty?
```
Comment (rolyatwilson) [inline spec/unit_test_spec.rb]: same as above

File: spec/unit_test_spec.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+require_relative 'helper/spec_helper'
+
+describe HockeyGerrit do
+  let(:hockey) { HockeyGerrit.new }
+
+  it '#git_log' do
+    log = hockey.git_log
```
Comment (rolyatwilson) [inline spec/unit_test_spec.rb]: why not "expect(hockey.git_log).to_not be_empty"

File: spec/unit_test_spec.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    raise 'git_log fail' if log.empty?
+  end
+
+  it '#git_commit_sha' do
+    log = hockey.git_commit_sha
```
Comment (rolyatwilson) [inline spec/unit_test_spec.rb]: same as above

File: spec/helper/spec_helper.rb (hunk 2 above)
Comment (bootstraponline) [inline spec/helper/spec_helper.rb]: good point, I'll make them constants.

File: spec/unit_test_spec.rb (hunk 5 above)
Comment (bootstraponline) [inline spec/unit_test_spec.rb]: Good point, fixed.

File: spec/unit_test_spec.rb (hunk 6 above)
Comment (bootstraponline) [inline spec/unit_test_spec.rb]: Fixed.

File: spec/hockey_gerrit_spec.rb [hunk 7]
```diff
@@ -1,5 +1,68 @@
...
+    allow(hockey).to receive(:git_log) { 'bootstraponline: Stub git log' }
+    allow(hockey).to receive(:git_commit_sha) { '123456789641603afe0572f3b4a91204791b012d' }
+  end
+
+  it 'posts app to hockey' do
+    stub_valid_response
+
+    hockey_run
+
+    expected_url = "#{RINK_HOCKEY_URL}/manage/apps/123456/app_versions/9"
```
Comment (rolyatwilson) [inline spec/hockey_gerrit_spec.rb]: /manage/apps/123456/app_versions/9
is used more than once. Can we promote it to a global in the spec_helper too?

**Comment by rolyatwilson** [comment]
Looks good to me

**Comment by bootstraponline** [comment]
merged!

**Pull request merged**